opb-crank/src/utils/utils.ts

```diff
@@ -0,0 +1,156 @@
+import * as os from 'os';
+import * as fs from 'fs';
+import {
+  AccountInfo,
+  Commitment,
+  Connection,
+  PublicKey,
+  PublicKeyInitData,
+} from '@solana/web3.js';
+
+import * as fzstd from 'fzstd';
+import { Market } from '@project-serum/serum';
+import * as bufferLayout from 'buffer-layout';
+import { Token, TOKEN_PROGRAM_ID } from '@solana/spl-token';
+
+export async function sleep(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+export function chunk(array, size) {
+  return Array.apply(0, new Array(Math.ceil(array.length / size)))
+    .map((_, index) => array.slice(index * size, (index + 1) * size));
+}
+
+export async function getMultipleAccounts(
+  connection: Connection,
+  publicKeys: PublicKey[],
+  commitment?: Commitment,
+  minContextSlot?: number,
+): Promise<{
+  publicKey: PublicKey;
+  context: { slot: number };
+  accountInfo: AccountInfo<Buffer>;
+}[]> {
+  if (!publicKeys.length) {
+    throw new Error('no Public Keys provided to getMultipleAccounts');
+  }
+
+  // set the maximum number of accounts per call
+  const chunkedPks = chunk(publicKeys, 100);
+
+  // asynchronously fetch each chunk of accounts and combine the results
+  return (await Promise.all(chunkedPks.map(async (pkChunk) => {
+    // use zstd to compress large responses
+    const encoding = 'base64+zstd';
+
+    const args = [pkChunk, { commitment, encoding, minContextSlot }];
+
+    // TODO: Use getMultipleAccounts()
+    // @ts-ignore
+    // eslint-disable-next-line no-underscore-dangle
+    const gmaResult = await connection._rpcRequest('getMultipleAccounts', args);
+
+    if (gmaResult.error) {
+      throw new Error(gmaResult.error.message);
+    }
+
+    return gmaResult.result.value.map(
+      ({
+        data, executable, lamports, owner,
+      }, i) => ({
+        publicKey: pkChunk[i],
+        context: gmaResult.result.context,
+        accountInfo: {
+          data: Buffer.from(fzstd.decompress(Buffer.from(data[0], 'base64'))),
+          executable,
+          owner: new PublicKey(owner),
+          lamports,
+        },
+      }),
+    );
+  }))).flat();
+}
+
+// load multiple markets at once instead of calling getAccountInfo
+// for each market 3 times by default it is 1 call to get the market
+// and 2 calls to get the decimals for baseMint and quoteMint
+// this can be condensed into 2 calls total per 100 markets
+export async function loadMultipleOpenbookMarkets(connection, programId, marketsList) {
+  const marketsMap = new Map();
+  const decimalMap = new Map();
+  const uniqueMints = new Set();
+
+  // get all the market data for an openbook market
+  const pubKeys = marketsList.map((item) => new PublicKey(item.address));
+  const marketsAccountInfos = await getMultipleAccounts(connection, pubKeys, 'processed');
```

Comment:
If we dont plan on doing more than 100 markets at once, then let's just directly use the rpc instead of the chunking helper.